### ISSUE TITLE
Turok 3 Incorrect Shadows

### ISSUE THREAD

**Issue by AmbientMalice** [gonetz/GLideN64#543]
I believe Turok 3 uses auxiliary framebuffer for shadows, whereas Turok 2 used stencil shadows. The shadows render better than Glide64, but they're too dark.

GLideN64:
![gliden64_turok_3 _shadow_of_o_017](https://cloud.githubusercontent.com/assets/10704701/7716521/643d2abe-fed6-11e4-9e56-8a3565196edd.jpg)

Angrylion:
![ntke0001](https://cloud.githubusercontent.com/assets/10704701/7716510/521d0e12-fed6-11e4-8568-3920680424c8.jpg)

**Comment by gonetz** [comment]
Savestate please.

**Comment by AmbientMalice** [comment]
https://www.sendspace.com/file/17np8s

**Comment by gonetz** [comment]
I guess I found how it works: manipulation with texture format again. Auxiliary buffer has format 8bit intensity, but when game use it as texture, texture loaded as 8bit intensity-alpha. Alpha makes the shadow translucent. I simulated this in shaders, but the shadow has strange boarder.
![gliden64_turok_3 _shadow_of_o_006](https://cloud.githubusercontent.com/assets/8983092/8235064/a0919ff8-1601-11e5-8166-5b6a7a0966a4.png)

**Comment by Predator82Germany** [comment]
& too dark

**Comment by gonetz** [comment]
Wrong intensity is a minor defect.

**Comment by AmbientMalice** [comment]
Does anti-aliasing have any effect on the softer shadows?

**Comment by gonetz** [comment]
No. I think, the shadow intentionally made "hairy". It looks like this with software lle too. Dark bound problem is probably solved by proper pixel coverage emulation, which is not implemented in GLideN64. I can't explain though, why shadow itself is darker than with soft lle.

**Comment by AmbientMalice** [comment]
Been testing https://github.com/gonetz/GLideN64/commit/d5a0f20098b838266485d211a5781e01a18dfec1

Glide64:
![glide64_jet_force_gemini_01](https://cloud.githubusercontent.com/assets/10704701/8252088/83638d46-16c5-11e5-9fb5-88c861430e51.png)

GLideN64:
![gliden64_jet_force_gemini_001](https://cloud.githubusercontent.com/assets/10704701/8252100/98fb85dc-16c5-11e5-8d88-439b49d221c9.jpg)

Angrylion's:
![njfe0000](https://cloud.githubusercontent.com/assets/10704701/8252113/bcce5afc-16c5-11e5-81a1-ea8bdc4107a0.jpg)

I'm not sure this half-fix was a good idea because it makes games that use this shadow method look really, really ugly. Even the inaccurate hard black shadows are, IMO, preferable to these harsh black fuzzing things.

**Comment by gonetz** [comment]
I reverted my fix. Need to find another solution.

**Comment by olivieryuyu** [comment]
![GLideN64_JET_FORCE_GEMINI_001](https://user-images.githubusercontent.com/8987490/109987975-5f2d9d00-7d07-11eb-9885-6196d54da092.png)

seems fixed
